### PR TITLE
chore(gcal): implement oauth

### PR DESCRIPTION
**Pull request by nickoferrall**
Fix https://github.com/ParabolInc/parabol/issues/8530

Demo: https://www.loom.com/share/f82541b86a9b49fe8ac0873fb49cffe6

I've applied to publish our Gcal App. In the meantime, I've added a bunch of people at Parabol as test users so we can test it internally: https://console.cloud.google.com/apis/credentials/consent?project=gcal-integration-386208

### To test

- [ ] Add the gcal user feature flag
- [ ] Add the `GCAL_CLIENT_ID` and `GCAL_CLIENT_SECRET` from the devops repo [here](https://github.com/ParabolInc/action-devops/blob/9cdf05e87e8ed9235f306a592ffef911b318ff0a/environments/local#L25-L26)
- [ ] Run `yarn build` then `yarn predeploy` to run `primeIntegrations`
- [ ] Can add & remove the gcal integration 
- [ ] Can create a gcal event in your calendar after integrating

### CODE REVIEW

**Review by igorlesnenko — APPROVED** [review]
Works! But some clean up required before merging.

**Comment by mattkrick** [comment]
Q: Can we get away with fewer permissions? Feels like the ability to delete someone's calendar is a bit of an overreach
image

**Comment by mattkrick** [comment]
Everything works! There are a couple UI kinks, but nothing blocking merge:
- Let me uncheck the box by clicking the label
- Show me an error message if you can't create the Gcal event
- Use calendar/time picker for dates (can kick this back to another date)

Loom here describing them: https://www.loom.com/share/025b4d6d67734a52afab73c04ddfdb21

**Review by mattkrick — CHANGES_REQUESTED** [review]
Very minor code stuff! Perms may be an issue, up to you when to fix

**Review by nickoferrall — COMMENTED** [review]
Updates:

- [ ] Refactored to sdl pattern
- [ ] You can check the checkbox by clicking on the label
- [ ] Created a GH issue for handling the error state here https://github.com/ParabolInc/parabol/issues/8635. This is just for internal testing so we can ship this PR without that. I also have a GH issue for improving the date UI & UX: https://github.com/ParabolInc/parabol/issues/8575
- [x] Reduced the amount of permissions we request 

Screenshot 2023-08-09 at 14 37 00

File: packages/server/graphql/public/typeDefs/TeamMemberIntegrations.graphql
Comment (mattkrick) [inline packages/server/graphql/public/typeDefs/TeamMemberIntegrations.graphql:4]: +1 Could you please remove `type TeamMemberIntegrations` from `_legacy.graphql` now that it lives in its own home?

**Review by mattkrick — APPROVED** [review]
awesome! love the smaller scope. 1 itty bitty fixup & then feel free to merge

**Pull request merged**
